src/components/CurrencyIcon/index.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useState } from 'react';
+
+import CurrencyPlaceholderIcon from '../../assets/img/coins/currency-placeholder.svg';
+import { useTypedSelector } from '../../redux/hooks';
+import { isLbtc, isLcad, isUsdt } from '../../utils/helpers';
+
+interface CurrencyIconProps {
+  assetHash: string;
+  size?: number;
+}
+
+export const CurrencyIcon = ({ assetHash, size = 24 }: CurrencyIconProps): JSX.Element => {
+  const [icon, setIcon] = useState<string | null>(null);
+  const network = useTypedSelector(({ settings }) => settings.network);
+
+  useEffect(() => {
+    (async () => {
+      let res: Response;
+      const reader = new FileReader();
+      // Display lbtc/usdt/lcad icons for all networks
+      if (isLbtc(assetHash, network)) {
+        res = await fetch(
+          `https://liquid.network/api/v1/asset/6f0279e9ed041c3d710a9f57d0c02928416460c4b722ae3457a11eec381c526d/icon`
+        );
+      } else if (isUsdt(assetHash, network)) {
+        res = await fetch(
+          `https://liquid.network/api/v1/asset/ce091c998b83c78bb71a632313ba3760f1763d9cfcffae02258ffa9865a37bd2/icon`
+        );
+      } else if (isLcad(assetHash, network)) {
+        res = await fetch(
+          `https://liquid.network/api/v1/asset/0e99c1a6da379d1f4151fb9df90449d40d0608f6cb33a5bcbfc8c265f42bab0a/icon`
+        );
+      } else {
+        res = await fetch(`https://liquid.network/api/v1/asset/${assetHash}/icon`);
+        if (!res.ok) {
+          setIcon(null);
+          return;
+        }
+      }
+      const imageBlob = await res.blob();
```

Comment:
Isn't this base64 serialization expensive?

Why not using a `<img src="">` component and pass the URL?

Reply:
Ha yes indeed. I did that for dashboard, to make it work with the AntD Icon component but doesn't apply here.